Cargo.toml

```diff
@@ -10,6 +10,7 @@ keywords = ["game", "engine", "gamedev", "graphics", "bevy"]
 license = "MIT OR Apache-2.0"
 readme = "README.md"
 repository = "https://github.com/bevyengine/bevy"
+rust-version = "1.65.0"
```

Comment:
Shouldn't this be specified on all of the crates, not just the top level `bevy`?

Reply:
I'm not sure. For the moment, the only crates that make sense to use individually would be `bevy_ecs` and `bevy_reflect`. There is no "need" to have them all on the same msrv, so that could mean multiplying the time in CI...

In my opinion the role of this field is mostly to provide a better error message for Bevy users.

If done with workspace inheritance, I would prefer to do most of the `Cargo.toml` fields at the same time, there are already at least one PR up for that

Reply:
I think it still makes sense from a plugin developer perspective, where we encourage depending on individual crates over the top level `bevy` to minimize compile times.

Reply:
> we encourage depending on individual crates over the top level `bevy`

We don't?

Reply:
Just checked the plugin guidelines and I guess we never did add those findings to it.  Issues like #5528 are going to get worse as the engine and ecosystem grows. IMO, discoverability is a weak argument for serializing dependency compilation on a single choke point, especially when alternatives for discoverability exist.

This isn't the place for this conversation though.
 
 > For the moment, the only crates that make sense to use individually would be bevy_ecs and bevy_reflect. There is no "need" to have them all on the same msrv, so that could mean multiplying the time in CI...
 
If these are the crates that are most avidly consuming new Rust language features, the rest of the engine is underpinned by both right now. If we bump the MSRV of either of them, the rest of the engine transitively takes on that MSRV as well, and with `bevy_reflect` being a dependency of `bevy_ecs`, the two are likely to be kept in sync. Likewise, if we want a MSRV on `bevy_ecs` , it should be on the crate itself, not the top level crate. I'm well aware of multiple groups picking and choosing parts of Bevy for their own purposes without pulling the whole engine in, and only putting it on `bevy` does not solve this issue for them.

Reply:
> I'm well aware of multiple groups picking and choosing parts of Bevy for their own purposes without pulling the whole engine in, and only putting it on bevy does not solve this issue for them.

Those would look like advanced users, at that point I would hope they are pinning versions of everything or vendoring, and have a good scenario for updates anyway.

I think the most interesting type of users for this check is new users of Bevy / Rust.
Putting it on all crates will be handled by workspace inheritance with #6089. I'll update this PR to use it if it's merged first, but I don't think this PR is the place to add it.

Reply:
Agreed with Francois here: we're targeting beginners with this change, who are overwhelmingly just going to use bevy itself.